### ISSUE TITLE
Bluetooth connection not working

### ISSUE THREAD

**Issue by octavmarius** [ev3dev/vscode-ev3dev-browser#85]
**Describe the bug**
I cannot connect from vs code to ev3 brick - device is not listed.

**To Reproduce**
1. start vs code
2. boot up ev3 brick from sd card 
3. enable bluetooth (check "Powered")
4. pear the brick with the computer
5. in vs code - ev3dev device browser - click on "Click here to connect to a device"

**Expected behavior**
expected to have my ev3 brick listed

**Desktop (please complete the following information):**
 - Windows 7 Proffesional  7601 SP1

**Additional context**
i can pear the device (ev3 brick) with my computer. 
i also removed (uninstalled) the ev3 brick from my computer and paired the device again with success.

![image](https://user-images.githubusercontent.com/28234605/56910852-3194be80-6ab4-11e9-9e7a-53a7edee8b84.png)

**Comment by haiyongsong1921** [comment]
@dlech Hi David, I have the same bluetooth connection issue in Mac OS. Does the ev3-browser extensions of the VSCode support for bluetooth connection. Because when I try to connect my ev3dev brick via bluetooth in VSCode, it will always throw out the handshake timeout error: "Failed to connect to ev3: Timed out while waiting for handshake."  But the connection works fine via usb cable. Would you mind to share some info for this? thanks a lot

**Comment by netanelbardavid** [comment]
@dlech Hi David, having the same issue with windows 10. Pairing works fine but the EV3 does not appear on the device list. Tried to connect it also by its IP address but VS Code shows the "Failed to connect to ev3: Timed out while waiting for handshake" error.
Thanks man

**Comment by jirklspot** [comment]
> @dlech Hi David, having the same issue with windows 10. Pairing works fine but the EV3 does not appear on the device list. Tried to connect it also by its IP address but VS Code shows the "Failed to connect to ev3: Timed out while waiting for handshake" error.
> Thanks man

Same here. Windows 10, Paired without any issue. but the device is not listing when I try to do ">ev3dev:Connect to a device" in Command Pallet. Any updates on this issue?

**Comment by dlech** [comment]
macOS and Windows 10 are not the same as the original issue (Windows 7), so please open a new issue with the exact steps to reproduce the problem and we will see if we can figure it out. Windows 10 and macOS are both working for me, so I need to know all of the details to see what is different.

**Comment by jyouturner** [comment]
FYI, I had the same problem on my Mac, but after I restarting Visual Code, I was able to connect to ev3dev from the extension/device list.

**Comment by jirklspot** [comment]
> macOS and Windows 10 are not the same as the original issue (Windows 7), so please open a new issue with the exact steps to reproduce the problem and we will see if we can figure it out. Windows 10 and macOS are both working for me, so I need to know all of the details to see what is different.

Thanks!  reported as new issue as you suggested >> https://github.com/ev3dev/vscode-ev3dev-browser/issues/97

**Comment by dlech** [comment]
>  Because when I try to connect my ev3dev brick via bluetooth in VSCode, it will always throw out the handshake timeout error: "Failed to connect to ev3: Timed out while waiting for handshake." But the connection works fine via usb cable.

In v1.1.0, the timeout is increased from 20 to 30 seconds. It is also now configurable via a setting if that is not long enough.

**Comment by vincentmohiuddin** [comment]
Hi, I was having the same issue. Pairing worked but the device would not show up in VSCode. It showed up in VSCode after I went to Wireless and Networks > All Network Connections > {PC-Name} > Connect on the brick.

**Comment by crucialfelix** [comment]
I am able to pair from Mac (Ventura) to ev3dev, and ev3dev shows that it's connected to my mac.  

But still VSCode EV3DEV DEVICE BROWSER shows it can't connect.

Oh wait ... I tried to reconnect (for like the 20th time today) and mysteriously it found  a newly labelled ev3dev and it works!

Oh I wish I could tell you why it works now and didn't before. Bluetooth is a mess

solved: it was because I also had the USB connected. That always works.

**Comment by petusky** [comment]
I have a similar experience as @crucialfelix.  I'm on Sequoia 15.3.  

I am able to pair my EV3 with my mac; HOWEVER, I can only connect FROM the mac TO the EV3, not the other way around. Here's evidence of such a bluetooth connection.

(Aside:  I called from Terminal, when i was not connected via USB, not from an SSH session. I only mention this because bluetooth "seems" to work this way when you have a simultaneous wired connection---as @curcialfelix notes, "solved: it was because I also had the USB connected. That always works."---but it's actually the wired connection that provides the IP address that lets you connect using the ev3dev extensions in Visual Studio Code).

```
APETUS-FMBP23-9:~ apetusky$ system_profiler SPBluetoothDataType
Bluetooth:

      Bluetooth Controller:
          Address: A4:CF:99:69:36:61
          State: On
          Chipset: BCM_4378B3
          Discoverable: Off
          Firmware Version: 22.1.143.281
          Product ID: 0x4A0F
          Supported services: 0x392039 < HFP AVRCP A2DP HID Braille LEA AACP GATT SerialPort >
          Transport: PCIe
          Vendor ID: 0x004C (Apple)
      Connected:
          ev3dev:
              Address: 00:17:EC:F5:AC:51
              Vendor ID: 0x0694
              Product ID: 0x0005
              Firmware Version: 3.1.6
              Minor Type: Computer
              RSSI: -32
              Services: 0x800018 < AVRCP A2DP ACL >
```

If I try to initiate the connection FROM the EV3, I get an error on the EV3 like "GDBus.Error:org:bluez.Error.Failed: Protocol not available".

The problem is that even though the devices are connected, the EV3 doesn't recognize this connection (which was established by initiating it on the Mac rather than the EV3) outside of the bluetooth window.  That is, EV3 knows it is connected:

<img width="429" alt="Image" src="https://github.com/user-attachments/assets/4792fca1-645b-47c7-8734-26f71de54a33" />

But in the Wireless and Networks screen it does not recognize the connection:

<img width="359" alt="Image" src="https://github.com/user-attachments/assets/c73db25f-ff11-4070-94ae-c7ffcd3215d4" />
  
As a result, no IP address shows up for the connection, as described in the last item in the insructions, https://www.ev3dev.org/docs/tutorials/connecting-to-the-internet-via-bluetooth/,

<img width="753" alt="Image" src="https://github.com/user-attachments/assets/63e6a1d3-67cf-4bd9-b5aa-9301aae8f0a8" />

However, you need this IP address to set up a connection with bluetooth in VSCode.  

So, what to do?  Does anyone know how to resolve the "GDBus.Error:org:bluez.Error.Failed: Protocol not available" error, which seems to be the source of the problem? I've updated just about everything I could update on my brick, but that error persists.

(BTW, when using the EV3 with EV3Classroom, I am still able to connect via bluetooth.  So I don't think it's an issue with the brick itself.)

**Comment by dlech** [comment]
This issue is about Windows, not Mac. There has been some more recent discussion about Mac at https://github.com/ev3dev/vscode-ev3dev-browser/issues/121

**Comment by petusky** [comment]
Agreed that my issue is a Mac issue.  But I don't think it's related to #121
<https://github.com/ev3dev/vscode-ev3dev-browser/issues/121>.  My problem
is outside of Visual Studio Code.  I can't establish a bluetooth connection
when I try to initiate it from the EV3, which I need to do to get the EV3
to recognize it and establish an IP Address to enter into Visual Studio
Code.

(Also, FYI, my mac already had Visual Studio Code selected as a local
network. So perhaps Microsoft already addressed that bug(?) )
